src/Pact/Types/Parser.hs

```diff
@@ -45,3 +64,87 @@ style = IdentifierStyle "atom"
         (HS.fromList ["true","false"])
         Symbol
         ReservedIdentifier
+
+-- -------------------------------------------------------------------------- --
+-- Pact Attoparsec backend parser
+
+-- | A wrapper around Attoparsec that adds DeltaParsing
+--
+newtype PactAttoparsec a = PactAttoparsec
+    { runPactAttoparsec :: StateT Int AP.Parser a }
+    deriving (Functor, Applicative, Alternative, Monad, MonadPlus, MonadState Int)
+    deriving (Parsing, TokenParsing)
+
+pactAttoParseOnly :: PactAttoparsec a -> T.Text -> Either String a
+pactAttoParseOnly = AP.parseOnly . flip evalStateT 0 . runPactAttoparsec
+
+-- | Atto DeltaParsing instance provides 'position' only (with no support for
+-- hidden chars like Trifecta).
+--
+instance DeltaParsing PactAttoparsec where
+    line = return mempty
+
+    -- Returns @Column NUMBER_OF_CHARS NUMBER_OF_BYTES@
+    --
+    -- Notes about legacy behavior:
+    --
+    -- For text <2 it used to return for both values the number of 16-bit code
+    -- units words. With the UTF-16 encoding used by text <2 this is the total
+    -- number of characters plus number of characters that are encoded using two
+    -- 16-bit code units. For instance the Ugaritic letter U+1038D would result
+    -- in a positional length of two (two 16-bit code units). The string
+    -- "a\0x263A\0x1038D" ("a☺𐎍") would have positional length 4 (3 characters
+    -- plus one 2 16-bit character).
+    --
+    -- In practice the old behavior was close enough to the number of characters
+    -- that it went mostly unnoticed and didn't cause harm on chain. The code
+    -- just assumed that it represented the number text characters. Those
+    -- numbers appear on chain (up to some block height) within info objects and
+    -- later still in failure messages. It is also relevant for extracting the
+    -- module text from the pact transaction before storing it in the pact db.
+    -- The presence of unicode characters can result in modules containing
+    -- dangling data because there are less characters in the module than what
+    -- is assumed based on the position information.
+    --
+    -- For text >=2 the attoparsic position tracks just bytes and the internal
+    -- representation of UTF-8. For instance the Ugaritic letter U+1038D results
+    -- in a byte length of 4. The string "a\0x263A\0x1038D" ("a☺𐎍") has 8 bytes
+    -- (1 code unit plus 3 code unit plus 4 code units).
+    --
+    position = do
+#if MIN_VERSION_text(2,0,0)
+        APT.Pos !bytePos <- parserPos
+        !charPos <- gets fromIntegral
+#elif LEGACY_PARSER == 1
+        APT.Pos !bytePos <- parserPos
+        let !charPos = fromIntegral bytePos
+#else
+        APT.Pos !bytePos <- (* 2) <$> parserPos
+        !charPos <- gets fromIntegral
+#endif
+        return $ TF.Columns charPos (fromIntegral bytePos)
+    {-# INLINE position #-}
+
+    slicedWith f a = (`f` mempty) <$> a
+    rend = return mempty
```

Comment:
Similar to `line`, I think `rend` and `restOfLine`, if not used, should return informative error messages